src/20231123_AaveV2Ethereum_RedeemCRVFromAaveV2EthereumAndTransferToGLC/AaveV2Ethereum_RedeemCRVFromAaveV2EthereumAndTransferToGLC_20231123.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+
+/**
+ * @title Redeem CRV from AaveV2Ethereum and Transfer to GLC
+ * @author efecarranza.eth
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xf92c5647c7f60a4a3db994b4953fc4408f5946cafdc0cebcd4c5924f40e04d36
+ * - Discussion: https://governance.aave.com/t/arfc-deploy-acrv-crv-to-vecrv/11628
+ */
+contract AaveV2Ethereum_RedeemCRVFromAaveV2EthereumAndTransferToGLC_20231123 is
+  IProposalGenericExecutor
+{
+  address public constant GLC_SAFE = 0x205e795336610f5131Be52F09218AF19f0f3eC60;
+
+  function execute() external {
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.CRV_A_TOKEN,
+      address(this),
+      IERC20(AaveV2EthereumAssets.CRV_A_TOKEN).balanceOf(address(AaveV2Ethereum.COLLECTOR))
+    );
+
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.CRV_A_TOKEN,
+      address(this),
+      IERC20(AaveV3EthereumAssets.CRV_A_TOKEN).balanceOf(address(AaveV3Ethereum.COLLECTOR))
+    );
+
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.CRV_UNDERLYING,
+      address(this),
```

Comment:
Does this work? Because the code uses the balance of `msg.sender` to withdraw.
I tried to update it and run tests and it reverts as well.

Reply:
i think comment is on wrong line, will comment at the right place - also plz resolve conflicts

Reply:
```suggestion
      GLC_SAFE,
```
yes it does work, you can also just try - you're tests are still passing with that change

Reply:
So we can,
- transfer v2 aCRV from collector to the executor
- transfer v3 aCRV from collector to the executor
- transfer underlying CRV from the collector to the GLC safe
- withdraw CRV from v2 pool directly to the GLC safe
- withdraw CRV from v3 pool directly to the GLC safe

not sure why it could revert.